### ISSUE TITLE
Remove Optional[] from all()

### ISSUE THREAD

**Issue by johnthagen** [collerek/ormar#174]
**Describe the bug**

why does `all()` return `List[Optional["T"]]` when the docs say:

```
If there are no rows meeting the criteria an empty list is returned.
```

Shouldn't that indicate that it returns `List["T"]` and it will just be an empty `[]` if it doesn't find anything? What does `None` communicate?

The `Optional` seems to flow form:

```py
    def _process_query_result_rows(self, rows: List) -> List[Optional["T"]]:
        """
        Process database rows and initialize ormar Model from each of the rows.

        :param rows: list of database rows from query result
        :type rows: List[sqlalchemy.engine.result.RowProxy]
        :return: list of models
        :rtype: List[Model]
        """
        result_rows = [
            self.model.from_row(
                row=row,
                select_related=self._select_related,
                excludable=self._excludable,
                source_model=self.model,
                proxy_source_model=self.proxy_source_model,
            )
            for row in rows
        ]
        if result_rows:
            return self.model.merge_instances_list(result_rows)  # type: ignore
        return cast(List[Optional["T"]], result_rows)
```

**Comment by johnthagen** [comment]
Task: https://github.com/collerek/ormar/pull/173#issuecomment-825351602

**Comment by collerek** [comment]
Fixed in 0.10.5 (638af9a) as I was fixing fastapi-pagination error you found.

**Comment by johnthagen** [comment]
I can confirm 0.10.5 solved this type checking issue. Thanks!
